ANES3-380: Image card variant

joegl opened the pull request:
# READY FOR REVIEW

# Summary
Implements the "Show Bottom Flourish" boolean field for card paragraphs in the Anesthesiology profile. Uninstalls the obsolete `anes_profile_helper` module and introduces a new `anes_helper` module under `modules/`.

# Review By (Date)
2025-10-24

# Urgency
Medium.

# Steps to Test

1. Sync and update local site.
3. Add or edit a card paragraph; verify the "Show Bottom Flourish" checkbox appears in the card behaviors form along with the default options from Stanford Profile Helper.


# Associated Issues and/or People
[ANES3-380](https://stanfordits.atlassian.net/browse/ANES3-380): BE | Configure Image Card variant

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)


[ANES3-380]: https://stanfordits.atlassian.net/browse/ANES3-380?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

joegl committed:
Remove anes_profile_helper.

joegl committed:
Restore anes_profile_helper for uninstall.

joegl committed:
Add update hook to uninstall anes_profile_helper.

joegl committed:
Added and enabled anes_helper module.

joegl committed:
Implement AnesCardBehaviors and add new Visual Chip Corner field.

joegl committed:
Remove test.

joegl committed:
Added empty line to module file.

pookmish commented on docroot/profiles/lagunita/anes_profile/anes_profile.install:
```diff
@@ -11,3 +11,12 @@
+/**
+ * Uninstall anes_profile_helper module.
+ */
+function anes_profile_update_11001() {
```
You don't really need this since it'll do it via config import. I find its best to avoid editing the profile files to make inheritance easier, les conflicts that way.

joegl replied:
Ahh, that's why I see the hooks in the `_helper` modules in the other profiles. Makes sense. I can ditch this and just rely on config import, but good to know.

pookmish commented on docroot/profiles/lagunita/anes_profile/config/sync/core.extension.yml:
```diff
@@ -247,7 +248,6 @@ module:
   menu_link_weight: 10
   stanford_profile_helper: 10
   paragraphs: 11
-  anes_profile_helper: 20
```
I'm on my phone so I can easily see.. did you leave the old info.yml file to avoid errors like "module does not exist"?

joegl replied:
Yup. That'll have to be removed later.

pookmish commented on docroot/profiles/lagunita/anes_profile/modules/anes_helper/anes_helper.module:
```diff
@@ -0,0 +1,13 @@
+/**
+ * Implements hook_paragraphs_behavior_info_alter().
+ */
+function anes_helper_paragraphs_behavior_info_alter(&$paragraphs_behavior) {
```
For a simple hook like this, I'm fine in the .module file for now. But we should get in the habit of using the class based hooks. Then we can include unit tests with them. 

FYI, I found GitHub copilot CLI does the test writing really easy. If you are in the root stack directory you can tell it to make a unit or kernel test. Then it'll even run it. But before it runs the test, run `drush phpunit:config` to set up the needed file in docroot/core/phpunit.xml. `vendor/bin/phpunit -c docroot/core docroot/path/to/tests`

joegl replied:
Good point. I'll go with the class hook.

pookmish reviewed (changes requested):
If you remove the update hook, I think this can be merged in. But as I mentioned I'm on my phone so review skills are limited.

rebeccahongsf commented:
Thank you for checking in, @joegl! Apologies for not specifying this in the ticket earlier. 

While reviewing designs with Kat last week, she mentioned that the client is more familiar with the term “Flourish” instead of “Chip” (totally understandable since we use the verbiage “Chip” to mean something else internally 😅).

> - What should the field be named and what should the description be?

Let's actually update the wording to
Label: “Show Bottom Flourish”
Help text: “Add a decorative visual element to the bottom of the card.”

> - Is there any other functionality you need, or just this field added to the card behaviors?

Would it be possible to remove the requirement for the Link name? That way, the user could leave the field blank to make the entire card function as a link instead.

Let me know your thoughts and if any additional info is needed. Thanks again!

joegl committed:
Merge branch '3.x' into ANES3-380--image-card-variant

joegl committed:
Removed database update hook.

joegl committed:
Moved procedural hook to a class hook.

joegl committed:
Update field name and description.:

joegl committed:
Unrequire the link text on cards.

joegl commented:
Ready for review again.

@pookmish I dropped the database update and moved the procedural hook to a class hook.

@rebeccahongsf I changed the field name and description. I also dropped the link text require on the URL field in the card. Just make note that this affects all Stanford Card paragraphs and that front-end handling of cards without link text is done gracefully 👍 Otherwise let me know and I can revert that change.

pookmish reviewed (approved):
LGTM

joegl committed:
Add missing dependency.
